test/TestUtilities/TestUtilities.jl

```diff
@@ -0,0 +1,197 @@
+    zlim = (0, 1)
+    domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(zlim[1]),
+        Geometry.ZPoint{FT}(zlim[2]);
+        boundary_tags = (:bottom, :top),
+    )
+    mesh = Meshes.IntervalMesh(domain, nelems = zelem)
+    topology = Topologies.IntervalTopology(context, mesh)
+    return Spaces.CenterFiniteDifferenceSpace(topology)
+end
+
+#= (single column) =#
+function ColumnFaceFiniteDifferenceSpace(
+    ::Type{FT};
+    zelem = 10,
+    context = ClimaComms.SingletonCommsContext(),
+) where {FT}
+    cspace = ColumnCenterFiniteDifferenceSpace(FT; zelem, context)
+    return Spaces.FaceFiniteDifferenceSpace(cspace)
+end
+
+function SphereSpectralElementSpace(
+    ::Type{FT};
+    context = ClimaComms.SingletonCommsContext(),
+) where {FT}
+    radius = FT(3)
+    ne = 4
+    Nq = 4
+    domain = Domains.SphereDomain(radius)
+    mesh = Meshes.EquiangularCubedSphere(domain, ne)
+    topology = Topologies.Topology2D(context, mesh)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    return Spaces.SpectralElementSpace2D(topology, quad)
+end
+
+function ExtrudedCenterFiniteDifferenceSpace(
+    ::Type{FT};
+    zelem = 10,
+    context = ClimaComms.SingletonCommsContext(),
+) where {FT}
+    radius = FT(128)
+    zlim = (0, 1)
+    helem = 4
+    Nq = 4
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(zlim[1]),
+        Geometry.ZPoint{FT}(zlim[2]);
+        boundary_tags = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = zelem)
+    vtopology = Topologies.IntervalTopology(context, vertmesh)
+    vspace = Spaces.CenterFiniteDifferenceSpace(vtopology)
+
+    hdomain = Domains.SphereDomain(radius)
+    hmesh = Meshes.EquiangularCubedSphere(hdomain, helem)
+    htopology = Topologies.Topology2D(context, hmesh)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    hspace = Spaces.SpectralElementSpace2D(htopology, quad)
+    return Spaces.ExtrudedFiniteDifferenceSpace(hspace, vspace)
+end
+
+function ExtrudedFaceFiniteDifferenceSpace(
+    ::Type{FT};
+    zelem = 10,
+    context = ClimaComms.SingletonCommsContext(),
+) where {FT}
+    cspace = ExtrudedCenterFiniteDifferenceSpace(FT; zelem, context)
+    return Spaces.ExtrudedFaceFiniteDifferenceSpace(cspace)
+end
+
+function all_spaces(
+    ::Type{FT};
+    zelem = 10,
+    context = ClimaComms.SingletonCommsContext(),
+) where {FT}
+    return [
+        PointSpace(FT),
+        SpectralElementSpace1D(FT; context),
+        SpectralElementSpace2D(FT; context),
+        # TODO: add these
+        # SpectralElementRectilinearSpace2D(FT; context),
+        # SpectralElementFiniteDifferenceRectilinearSpace2D(FT; context),
+        ColumnCenterFiniteDifferenceSpace(FT; zelem, context),
+        ColumnFaceFiniteDifferenceSpace(FT; zelem, context),
+        SphereSpectralElementSpace(FT; context),
+        ExtrudedCenterFiniteDifferenceSpace(FT; zelem, context),
+        # ExtrudedFaceFiniteDifferenceSpace(FT; zelem, context), # errors on sum
```

Comment:
@sriharshakandala, @simonbyrne, sorry, I realize now that I squashed this into the previous commit. Please uncomment to reproduce the error